--- a/src/syscall/sysnums.list
+++ b/src/syscall/sysnums.list
@@ -72,6 +72,7 @@
 SYSNUM(fchdir)
 SYSNUM(fchmod)
 SYSNUM(fchmodat)
+SYSNUM(fchmodat2)
 SYSNUM(fchown)
 SYSNUM(fchown32)
 SYSNUM(fchownat)
--- a/src/syscall/sysnums-arm64.h
+++ b/src/syscall/sysnums-arm64.h
@@ -281,4 +281,5 @@
 	[ 435 ] = PR_clone3,
 	[ 437 ] = PR_openat2,
 	[ 439 ] = PR_faccessat2,
+	[ 452 ] = PR_fchmodat2,
 };
--- a/src/syscall/enter.c
+++ b/src/syscall/enter.c
@@ -2483,6 +2483,7 @@
 			maybe_redirect_userns_file(tracee, SYSARG_1);
 		break;
 
+	case PR_fchmodat2:
 	case PR_fchownat:
 	case PR_fstatat64:
 	case PR_newfstatat:
--- a/src/syscall/seccomp.c
+++ b/src/syscall/seccomp.c
@@ -356,6 +356,7 @@
 	{ PR_faccessat2,	FILTER_SYSEXIT },
 	{ PR_fchdir,		FILTER_SYSEXIT },
 	{ PR_fchmodat,		0 },
+	{ PR_fchmodat2,		0 },
 	{ PR_fchownat,		0 },
 	{ PR_fstatat64,		0 },
 	{ PR_futimesat,		0 },
--- a/src/extension/fake_id0/fake_id0.c
+++ b/src/extension/fake_id0/fake_id0.c
@@ -321,6 +321,7 @@
 	{ PR_execve,		FILTER_SYSEXIT },
 	{ PR_fchmod,		FILTER_SYSEXIT },
 	{ PR_fchmodat,		FILTER_SYSEXIT },
+	{ PR_fchmodat2,		FILTER_SYSEXIT },
 	{ PR_fchown,		FILTER_SYSEXIT },
 	{ PR_fchown32,		FILTER_SYSEXIT },
 	{ PR_fchownat,		FILTER_SYSEXIT },
@@ -446,6 +447,7 @@
 			break;
 
 		case PR_fchmodat:
+	case PR_fchmodat2:
 			node->mode = peek_reg(tracee, ORIGINAL, SYSARG_3);
 			break;
 
@@ -642,6 +644,7 @@
 			SYSARG_1, IGNORE_SYSARG, config);
 	/* int fchmodat(int dirfd, const char *pathname, mode_t mode, int flags (unused)) */
 	case PR_fchmodat:
+	case PR_fchmodat2:
 		return handle_chmod_enter_end(tracee, SYSARG_2, SYSARG_3, 
 			IGNORE_SYSARG, SYSARG_1, config);
 
@@ -1035,6 +1038,7 @@
 	case PR_fchown32:
 	case PR_lchown32:
 	case PR_fchmodat:
+	case PR_fchmodat2:
 	case PR_fchownat: 
 		return handle_perm_err_exit_end(tracee, config, false);
 
